kyuubi-server/src/main/scala/org/apache/kyuubi/engine/KubernetesApplicationOperation.scala

```diff
@@ -167,6 +167,29 @@ class KubernetesApplicationOperation extends ApplicationOperation with Logging {
       TimeUnit.MILLISECONDS)
     cleanupCanceledAppPodExecutor = ThreadUtils.newDaemonCachedThreadPool(
       "cleanup-canceled-app-pod-thread")
+    initializeKubernetesClient(kyuubiConf)
+  }
+
+  private[kyuubi] def getKubernetesClientInitializeInfo(
+      kyuubiConf: KyuubiConf): Seq[KubernetesInfo] = {
+    kyuubiConf.get(KyuubiConf.KUBERNETES_CLIENT_INITIALIZE_LIST).map { init =>
+      val (context, namespace) = init.split(":") match {
+        case Array(ctx, ns) => (Some(ctx).filterNot(_.isEmpty), Some(ns).filterNot(_.isEmpty))
+        case Array(ctx) => (Some(ctx).filterNot(_.isEmpty), None)
+        case _ => (None, None)
+      }
+      KubernetesInfo(context, namespace)
+    }
+  }
+
+  private[kyuubi] def initializeKubernetesClient(kyuubiConf: KyuubiConf): Unit = {
+    getKubernetesClientInitializeInfo(kyuubiConf).foreach { kubernetesInfo =>
+      try {
+        getOrCreateKubernetesClient(kubernetesInfo)
+      } catch {
+        case e: Throwable => error(s"Failed to initialize Kubernetes client for $kubernetesInfo", e)
```

Comment:
this also helps the user to identify the configuration issues, I consistently getting reports from some users complain their server can not get engine status for Spark apps running on K8s, due to wrong conf cause server can not instance K8s Client correctly.